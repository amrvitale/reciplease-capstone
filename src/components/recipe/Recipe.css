* {
  box-sizing: border-box;
}

.container {
  display: flex;
  flex-direction: row;
  flex-flow: row wrap;
  align-items: center;
  justify-content: space-around;
}

.card { 
  margin: 1px;
  padding: 5px;
  border: 1px solid #2B2D42;
  border-radius: 5px;
  background-image: url('./recipecard2.jpg');
  justify-content: space-evenly;
}

.container .card {
  padding: 3px;
  display: flex;
  border: 1px #2B2D42;  
}

li.card{
  margin: 30px;
  justify-self: center;
  height: 250px;
  width: 320px;
  box-shadow: 3px 3px 5px 6px #ccc;
}
 

 .recipeTitle{
   color: rgb(19, 16, 16);
  padding: 6px;
  margin: 8px;
   display:inline-block;
   white-space: pre-wrap;
   overflow: hidden;
   width: 8em;
   -webkit-line-clamp: 2; /* number of lines to show */
   -webkit-box-orient: vertical;
  }
  
