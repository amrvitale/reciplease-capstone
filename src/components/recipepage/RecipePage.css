.recipeName {
    color: maroon;
    font-size: 40px;
    background-color: black;
    text-transform: uppercase;
}

p {
    white-space: pre-wrap;
}

