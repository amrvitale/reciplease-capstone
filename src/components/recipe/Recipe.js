import React from 'react';
import { Link } from 'react-router-dom';
import './Recipe.css';
import ApiContext from '../../ApiContext';

/* This component renders the Recipe card. It is called by the My Recipes and Search components. */
export default class Recipe extends React.Component {
   
    static contextType = ApiContext
    
    render() {
        const {id, recipename, experience, preptime, cooktime, servings, ingredients, directions} = this.props
        
        return (
            <div className="recipeCard">
                <Link to={`/recipe/${id}`}>
                    <div className="card-header">
                        <div className="card-title">
                            <h2 className="recipeTitle">{recipename}</h2>
                        </div>
                    </div>           
                </Link>
            </div>
        ); 
    }
}