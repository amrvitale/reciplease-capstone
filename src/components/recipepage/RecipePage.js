import React from 'react';
import ApiContext from '../../ApiContext';
import config from '../../config';
import {Link} from 'react-router-dom';
import './RecipePage.css';

/* This component renders the page for a selected Recipe. Users can edit and publish a recipe on this page. */

class RecipePage extends React.Component {
    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = ApiContext

     updateRecipeStatus = (selectedRecipe) =>  {
        console.log('clicked')
        console.log(this.props)
    
        let status 
          
        if (selectedRecipe.status === 'published') {
          status = 'unpublished'
        } 

        else {
          status = 'published'
        }  
    
        const updatedRecipe = {
          ...selectedRecipe,
          status: status
        }
    
        let url = `${config.API_ENDPOINT}/myrecipes/${selectedRecipe.id}`; 
    
        fetch(url, {
            method: 'PUT',
            headers: {
                'content-type': 'application/json'
            },   
            body: JSON.stringify(updatedRecipe),
        }) 
    
        .then(recipe => {
          console.log(recipe)
          this.props.handlePublishRecipe(selectedRecipe.id, status);
        })
    
        .catch(error => {
          console.log({ error })
        })
      }

    render() {
        const {recipes = [] } = this.context
        console.log(this.context)
        const {slug} = this.props.match.params;
        console.log(slug)
        console.log(recipes)
        let selectedRecipe = recipes.find(recipe => recipe.id === parseInt(slug))
        console.log(recipes, selectedRecipe, "recipes", "selectedRecipe")

        let statusButton;
        
        if (selectedRecipe === undefined) {
            let html = <p>Readying your recipe!</p>
            return html
        }
        
        else if (selectedRecipe.status === "published") {
            statusButton = <button className="unpubButton" onClick= {() => this.updateRecipeStatus(selectedRecipe)}>Unpublish</button>
        }

        else {
            statusButton = <button className="pubButton" onClick= {() => this.updateRecipeStatus(selectedRecipe)}>Publish</button>
        }

        return (
            <div className="recipePage">
                <div className="recipeActions">
                    <Link to={`/editrecipe/${slug}`}><button type="button">Edit</button></Link>
                    {statusButton}
                </div>

                <div className="selectedRecipe">
                    <h1 className="recipeName">{(selectedRecipe) ? selectedRecipe.recipename: ""}</h1>
                    <br />
                    <p>Experience:</p>
                    <p className="experience">{(selectedRecipe) ? selectedRecipe.experience: ""}</p>
                    <br />
                    <p className="preptime">Prep time: {(selectedRecipe) ? selectedRecipe.preptime: ""}</p>
                    <br />
                    <p className="cooktime">Cook time: {(selectedRecipe) ? selectedRecipe.cooktime: ""}</p>
                    <br />
                    <p className="servings">Servings: {(selectedRecipe) ? selectedRecipe.servings: ""}</p>
                    <br />
                    <p>Ingredients:</p>
                    <p className="ingredients">{(selectedRecipe) ? selectedRecipe.ingredients: ""}</p>
                    <br />
                    <p>Directions:</p>
                    <p className="directions">{(selectedRecipe) ? selectedRecipe.directions: ""}</p>
                </div>
            </div>
        );
    }
}

export default RecipePage;