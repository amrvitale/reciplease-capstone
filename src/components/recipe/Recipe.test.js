import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Recipe from './Recipe'
import RecipePage from '../recipepage/RecipePage';

describe (`Recipe component`, () => {
    it ('renders a .Recipe by default', () => {
        const wrapper = shallow( <Recipe />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});