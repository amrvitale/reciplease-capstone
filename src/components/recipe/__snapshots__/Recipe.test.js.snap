// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component renders a .Recipe by default 1`] = `
<div
  className="recipeCard"
>
  <Link
    to="/recipe/undefined"
  >
    <div
      className="card-header"
    >
      <div
        className="card-title"
      >
        <h2
          className="recipeTitle"
        />
      </div>
    </div>
  </Link>
</div>
`;
